[native pos] Re-launch the native process if it's already crashed

miaoever opened the pull request:
We're reusing the native process within the JVM for different tasks, if the native process is crashed by previous task, the next coming task should re-launch it to prevent further failure.

```
== NO RELEASE NOTE ==
```

mbasmanova reviewed (approved):
@miaoever Looks good % typo in commit message. "Re=launch" -> "Re-launch". Also, please, add [native pos] prefix to commit title. I wonder if it is possible to add a test to ensure this behavior doesn't get broken by future changes.

miaoever commented:
> @miaoever Looks good % typo in commit message. "Re=launch" -> "Re-launch". Also, please, add [native pos] prefix to commit title. I wonder if it is possible to add a test to ensure this behavior doesn't get broken by future changes.

Fixed the title and added a test to cover the case.

mbasmanova reviewed (commented):
@miaoever Thank you for adding a test.
